ferry/importer_pandas.py

```diff
@@ -0,0 +1,209 @@
+
+        demand_file = Path(f"{config.DATA_DIR}/demand_stats/{season}_demand.json")
+
+        if not demand_file.is_file():
+            print(f"Skipping season {season}: demand statistics file not found.")
+            continue
+
+        with open(demand_file, "r") as f:
+            demand_info = pd.read_json(f)
+
+        demand_info["season_code"] = season
+        merged_demand_info.append(demand_info)
+
+    merged_demand_info = pd.concat(merged_demand_info, axis=0)
+    merged_demand_info = merged_demand_info.reset_index(drop=True)
+
+    demand_statistics = import_demand(merged_demand_info, listings, demand_seasons)
+
+    # -------------------------
+    # Import course evaluations
+    # -------------------------
+
+    print("\n[Importing course evaluations]")
+
+    # list available evaluation files
+    previous_eval_files = Path(config.DATA_DIR / "previous_evals").glob("*.json")
+    new_eval_files = Path(config.DATA_DIR / "course_evals").glob("*.json")
+
+    previous_eval_files = [x.name for x in previous_eval_files]
+    new_eval_files = [x.name for x in new_eval_files]
+
+    all_evals = sorted(list(set(previous_eval_files + new_eval_files)))
+
+    merged_evaluations_info = []
+
+    for filename in tqdm(all_evals, desc="Loading evaluation JSONs"):
+        # Read the evaluation, giving preference to current over previous.
+        current_evals_file = Path(f"{config.DATA_DIR}/course_evals/{filename}")
+
+        if current_evals_file.is_file():
+            with open(current_evals_file, "r") as f:
+                evaluation = ujson.load(f)
+        else:
+            with open(f"{config.DATA_DIR}/previous_evals/{filename}", "r") as f:
+                evaluation = ujson.load(f)
+
+        merged_evaluations_info.append(evaluation)
+
+    merged_evaluations_info = pd.DataFrame(merged_evaluations_info)
+
+    (
+        evaluation_narratives,
+        evaluation_ratings,
+        evaluation_statistics,
+        evaluation_questions,
+    ) = import_evaluations(merged_evaluations_info, listings)
+
+    # ------------------------
+    # Replace tables in database
+    # ------------------------
+    print("\n[Clearing database]")
+
+    # drop old tables
+    meta = MetaData(bind=database.Engine, reflect=True)
+    conn = database.Engine.connect()
+    delete = conn.begin()
+    for table in meta.sorted_tables:
+        conn.execute(f'ALTER TABLE "{table.name}" DISABLE TRIGGER ALL;')
+        conn.execute(table.delete())
+        conn.execute(f'ALTER TABLE "{table.name}" ENABLE TRIGGER ALL;')
+    delete.commit()
+
+    print("\n[Updating database]")
+
+    conn = psycopg2.connect(
+        host="localhost",
+        database="postgres",
+        user="postgres",
+        password="thisisapassword",
+    )
+
+    # insert new tables
+    seasons = pd.DataFrame(course_seasons, columns=["season_code"], dtype=int)
+    seasons["term"] = np.nan
+    seasons["year"] = np.nan
```

Comment:
wat

Reply:
I had to restore the seasons table, but since the term and year are generated in computed.py I just left these as-is for now

Reply:
Probably better to generate it here and then overwrite via computed.py. Eventually we can switch computed to just check this as an invariant.